src/seraphis_main/contextual_enote_record_types.cpp

```diff
@@ -86,20 +86,67 @@ rct::xmr_amount amount_ref(const SpContextualEnoteRecordV1 &record)
     return record.record.amount;
 }
 //-------------------------------------------------------------------------------------------------------------------
-const SpEnoteOriginContextV1& origin_context_ref(const ContextualBasicRecordVariant &variant)
+const EnoteOriginContextVariant& origin_context_ref(const ContextualBasicRecordVariant &variant)
 {
-    struct visitor final : public tools::variant_static_visitor<const SpEnoteOriginContextV1&>
+    // TODO: figure out how deeply I need to understand how variant and visitor work
+    struct visitor final : public tools::variant_static_visitor<const EnoteOriginContextVariant&>
     {
         using variant_static_visitor::operator();  //for blank overload
-        const SpEnoteOriginContextV1& operator()(const LegacyContextualBasicEnoteRecordV1 &record) const
+        const EnoteOriginContextVariant& operator()(const LegacyContextualBasicEnoteRecordV1 &record) const
         { return record.origin_context; }
-        const SpEnoteOriginContextV1& operator()(const SpContextualBasicEnoteRecordV1 &record) const
+        const EnoteOriginContextVariant& operator()(const SpContextualBasicEnoteRecordV1 &record) const
         { return record.origin_context; }
     };
 
     return variant.visit(visitor{});
 }
```

Comment:
Returning a reference to local variable will cause segfaults or other undefined behavior. https://stackoverflow.com/questions/4643713/returning-a-reference-to-a-local-variable-in-c

Reply:
Might be missing something -- is this definitely returning a reference to a local variable? `visitor` isn't returned (in case that's what you were pointing out), it's passed into `visit()`, which returns a reference to a member of the higher scoped `variant`

Reply:
The `operator()()` visitor member functions are constructing `EnoteOriginContextVariant` (a `tools::variant<...>`) from a `const` reference to `SpEnoteOriginContextV1` and `LegacyEnoteOriginContextV1`. Then the member functions are returning a `const` reference to that locally constructed variant.

Reply:
Ah, I see. You're right. Just noticed this is the compiler warning highlighted in the other part of the PR too.

Maybe the simplest solution is a return by reference here.